samples/nrf9160/nrf_cloud_rest_fota/src/main.c

```diff
@@ -0,0 +1,821 @@
+	int err = init_led();
+
+	if (err) {
+		LOG_ERR("LED initialization failed");
+		return err;
+	}
+	err = settings_subsys_init();
+	if (err) {
+		LOG_ERR("Failed to initialize settings subsystem, error: %d", err);
+		return err;
+	}
+	err = settings_load_subtree(settings_handler_rest_fota.name);
+	if (err) {
+		LOG_WRN("Failed to load settings, error: %d", err);
+	}
+
+	modem_lib_init_result = nrf_modem_lib_init(NORMAL_MODE);
+
+	/* This function may perform a reboot if a FOTA update is in progress */
+	process_pending_job(modem_lib_init_result);
+
+	if (modem_lib_init_result) {
+		LOG_ERR("Failed to initialize modem library: 0x%X", modem_lib_init_result);
+		return -EFAULT;
+	}
+
+	err = at_cmd_init();
+	if (err) {
+		LOG_ERR("Failed to initialize AT commands, error: %d", err);
+		return err;
+	}
+
+	err = at_notif_init();
+	if (err) {
+		LOG_ERR("Failed to initialize AT notifications, error: %d", err);
+		return err;
+	}
+
+	err = fota_download_init(http_fota_dl_handler);
+	if (err) {
+		LOG_ERR("Failed to initialize FOTA download, error: %d", err);
+		return err;
+	}
+
+	err = modem_info_init();
+	if (err) {
+		LOG_WRN("Modem info initialization failed, error: %d", err);
+		return err;
+	}
+
+	err = modem_info_params_init(&mdm_param);
+	if (!err) {
+		LOG_INF("App: %s, ver: %s",
+			log_strdup(mdm_param.device.app_name),
+			log_strdup(mdm_param.device.app_version));
+
+		err = modem_info_params_get(&mdm_param);
+		if (!err) {
+			LOG_INF("Modem FW Ver: %s",
+				log_strdup(mdm_param.device.modem_fw.value_string));
+		} else {
+			LOG_WRN("Unable to obtain modem info, error: %d", err);
+		}
+	} else {
+		LOG_WRN("Modem info params initialization failed, error: %d", err);
+	}
+
+	err = nrf_cloud_client_id_get(device_id, sizeof(device_id));
+	if (err) {
+		LOG_ERR("Failed to set device ID, error: %d", err);
+		return err;
+	}
+
+	LOG_INF("Device ID: %s", log_strdup(device_id));
+
+	err = dk_buttons_init(button_handler);
+	if (err) {
+		LOG_ERR("Failed to initialize button: error: %d", err);
+		return err;
+	}
+
+#if defined(CONFIG_REST_FOTA_DO_JITP)
+	/* Present option for JITP via REST */
+	request_jitp();
+#endif
+
+	/* Present option to enable FOTA in device shadow via REST */
+	request_fota_enable();
+
+	return 0;
+}
+
+static void lte_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS:
+		LOG_DBG("LTE_LC_EVT_NW_REG_STATUS: %d", evt->nw_reg_status);
+		if ((evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_HOME) &&
+		     (evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_ROAMING)) {
+			break;
+		}
+
+		LOG_DBG("Network registration status: %s",
+			evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME ?
+			"Connected - home network" : "Connected - roaming");
+		k_sem_give(&lte_connected);
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		LOG_DBG("LTE cell changed: Cell ID: %d, Tracking area: %d",
+			evt->cell.id, evt->cell.tac);
+		break;
+	case LTE_LC_EVT_NEIGHBOR_CELL_MEAS:
```

Comment:
What's the reason to leave these unused cases here?  Why not just default:?

Reply:
no particular reason other than to display the possible events.
i'm fine with removing them

Reply:
It's similar to the debate over whether to keep disabled but useful config parameters commented out in prj.conf.  Some folks like them because they help users see the possibilities, and others think it is just clutter.  I think the cleaning clutter people are winning.